File: benchmarks/bench_sum.zig
const std = @import("std");

fn sum(n: i32) i32 {
    var s: i32 = 0;
    var i: i32 = 1;
    while (i <= n) : (i += 1) {
        s += i;
    }
    return s;
}

pub fn main() u8 {
    var r: i32 = 0;
    var j: i32 = 0;
    while (j < 10000000) : (j += 1) {
        r = sum(100);
    }
    return @intCast(r & 0xFF);
}

File: benchmarks/bench_sum.c
#include <stdio.h>
#include <stdint.h>

static int32_t sum(int32_t n) {
    int32_t s = 0;
    for (int32_t i = 1; i <= n; i++) s += i;
    return s;
}

int main(void) {
    volatile int32_t r = 0;
    for (int i = 0; i < 10000000; i++) {
        r = sum(100);
    }
    return (int)(r & 0xFF);
}
